added security-Boot-feature

serhii-shyian opened the pull request.

serhii-shyian committed:
added security-Boot-feature

Elena-Bruyako commented on src/main/java/com/example/bookstore/model/User.java:
```diff
@@ -33,7 +42,49 @@ public class User {
+
+    @Override
+    public boolean isAccountNonExpired() {
+        return UserDetails.super.isAccountNonExpired();
```
```suggestion
        return true;
```

Elena-Bruyako commented on src/main/java/com/example/bookstore/model/User.java:
```diff
@@ -33,7 +42,49 @@ public class User {
+
+    @Override
+    public boolean isEnabled() {
+        return UserDetails.super.isEnabled();
```
```suggestion
        return !isDeleted;
```

Elena-Bruyako commented on src/main/java/com/example/bookstore/model/User.java:
```diff
@@ -33,7 +42,49 @@ public class User {
+
+    @Override
+    public boolean isCredentialsNonExpired() {
+        return UserDetails.super.isCredentialsNonExpired();
```
```suggestion
        return true;
```

Elena-Bruyako commented on src/main/java/com/example/bookstore/model/User.java:
```diff
@@ -33,7 +42,49 @@ public class User {
+
+    @Override
+    public boolean isAccountNonLocked() {
+        return UserDetails.super.isAccountNonLocked();
```
```suggestion
        return true;
```

Elena-Bruyako commented on src/main/java/com/example/bookstore/model/User.java:
```diff
@@ -33,7 +42,49 @@ public class User {
+            joinColumns = @JoinColumn(name = "user_id"),
+            inverseJoinColumns = @JoinColumn(name = "role_id"))
+    @ToString.Exclude
+    private Set<Role> roles;
```
```suggestion
    private Set<Role> roles = new HashSet<>();
```

Elena-Bruyako commented on src/main/java/com/example/bookstore/model/User.java:
```diff
@@ -33,7 +42,49 @@ public class User {
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return new HashSet<>(roles);
```
```suggestion
        return roles;
```

Elena-Bruyako commented on src/main/java/com/example/bookstore/security/CustomUserDetailsService.java:
```diff
@@ -0,0 +1,22 @@
+    public UserDetails loadUserByUsername(String email)
+            throws UsernameNotFoundException {
+        return userRepository.findByEmail(email).orElseThrow(
+                () -> new UsernameNotFoundException("Can't find user by email: " + email)
```
```suggestion
                () -> new EntityNotFoundException("Can't find user by email: " + email)
```

Elena-Bruyako reviewed (changes requested):
Good job! Let's improve your solution!

serhii-shyian committed:
1) Change UserNotFoundException to EntityNotFoundException in CustomUserDetailsService class
2) Replace creation new Hashset of roles from getAuthorities() method to roles field, set true return value to isAccountNonExpired(), isAccountNonLocked(), isCredentialsNonExpired() methods, set !isDeleted return value to isEnabled() method

Elena-Bruyako reviewed (approved):
well done
